trace/opencensus/trace/reporters/zipkin_reporter.py

```diff
@@ -0,0 +1,158 @@
+DEFAULT_PORT = 9411
+ZIPKIN_HEADERS = {'Content-Type': 'application/json'}
+
+ISO_DATETIME_REGEX = '%Y-%m-%dT%H:%M:%S.%fZ'
+
+SPAN_KIND_MAP = {
+    0: None,  # span kind unspecified
+    1: "SERVER",
+    2: "CLIENT",
+}
+
+SUCCESS_STATUS_CODE = (200, 202)
+
+
+class ZipkinReporter(object):
+    """Report the spans to Zipkin.
+
+    See: http://zipkin.io/zipkin-api/#
+
+    :type service_name: str
+    :param service_name: Service that logged an annotation in a trace.
+                         Classifier when query for spans.
+
+    :type host_name: str
+    :param host_name: (Optional) The host name of the Zipkin server.
+
+    :type port: int
+    :param port: (Optional) The port of the Zipkin server.
+
+    :type end_point: str
+    :param end_point: (Optional) The path for the span reporting endpoint.
+    """
+
+    def __init__(
+            self,
+            service_name,
+            host_name=DEFAULT_HOST_NAME,
+            port=DEFAULT_PORT,
+            endpoint=DEFAULT_ENDPOINT):
+        self.service_name = service_name
+        self.host_name = host_name
+        self.port = port
+        self.endpoint = endpoint
+        self.url = self.get_url
+
+    @property
+    def get_url(self):
+        return 'http://{}:{}{}'.format(
+            self.host_name,
+            self.port,
+            self.endpoint)
+
+    def report(self, trace):
+        """Send trace to Zipkin server, default using the v1 API.
+
+        :type trace: dict
+        :param trace: Trace data in dictionary format.
+        """
+        trace_id = trace.get('traceId')
+        spans = trace.get('spans')
+
+        try:
+            zipkin_spans = self.translate_to_zipkin(trace_id, spans)
+            result = requests.post(
+                url=self.url,
+                data=json.dumps(zipkin_spans),
+                headers=ZIPKIN_HEADERS)
+
+            if result.status_code not in SUCCESS_STATUS_CODE:
+                logging.error(
+                    "Failed to send spans to Zipkin server! Spans are {}"
+                    .format(zipkin_spans))
+        except Exception as e:  # pragma: NO COVER
+            logging.error(e.message)
+
+    def translate_to_zipkin(self, trace_id, spans):
+        """Translate the opencensus spans to zipkin spans.
+
+        :type trace_id: str
+        :param trace_id: Trace ID.
+
+        :type spans: list
+        :param spans: List of spans to be reported.
+
+        :rtype: list
+        :returns: List of zipkin format spans.
+        """
+        local_endpoint = {
+            'serviceName': self.service_name,
+            'ipv4': self.host_name,
+            'port': self.port,
+        }
+
+        zipkin_spans = []
+
+        for span in spans:
+            # Timestamp in zipkin spans is int of microseconds.
+            start_datetime = datetime.datetime.strptime(
+                span.get('startTime'),
+                ISO_DATETIME_REGEX)
+            start_timestamp = calendar.timegm(
```

Comment:
I suggest renaming `start_timestamp`, `end_timestamp`, and `duration` to `start_timestamp_ms`, `end_timestamp_ms` and `duration_ms`.